foursight_core/app_utils.py

```diff
@@ -256,7 +257,7 @@ def set_user_record(self, email: str, record: dict, error: str, exception: str):
             user_record["error"] = error
             user_record["exception"] = exception
 
-    def get_portal_url(self, env_name: str) -> str:
+    def get_portal_url(self, env_name: str) -> Optional[str]:
```

Comment:
The utility of raising an error (active error handling) is that the caller doesn't have to know about the error cases. Returning `None` is a passive form of error handling. I looked around for a caller of this and the first one I found (maybe there are others, too) doesn't seem to be prepared to receive `None`.
```
def get_auth0_client_id_from_portal(self, env_name: str) -> str:
        logger.warning(f"Fetching Auth0 client ID from portal.")
        portal_url = self.get_portal_url(env_name)
        auth0_config_url = portal_url + "/auth0_config?format=json"
```
This is going to end up raising an inscrutible `TypeError`.

Reply:
I'm falling back to the hardcoded value (as it previously was). Tightened to to make sure it returns something.